gammapy/modeling/models/spectral.py

```diff
@@ -183,6 +183,44 @@ def integral_error(self, emin, emax):
         flux_err = flux * dnde_err / dnde
         return u.Quantity([flux.value, flux_err.value], unit=flux.unit)
 
+    def _propagate_error(self, fct, emin, emax, epsilon=0.01):
```

Comment:
Can you take a look here: https://github.com/gammapy/gammapy/blob/master/gammapy/modeling/models/spectral.py#L139 `eps` is meant to be an array of values, as it is adapted to the value of the parameter. Can you change the implementation to take this into account?